hw7 is done

Ltaudato1 opened the pull request.

Ltaudato1 committed:
hw7 is done

Ltaudato1 committed:
hw8 is done

Ltaudato1 committed:
hw9

damir2407 commented on scrapper/src/main/java/edu/java/clients/StackOverflowClient.java:
```diff
@@ -20,6 +23,8 @@ public StackOverflowClient(WebClient.Builder webClientBuilder, String baseUrl) {
         this.webClient = webClientBuilder.baseUrl(baseUrl).build();
     }
 
+    @Retryable(value = {HttpStatusCodeException.class}, maxAttemptsExpression = "${retry.maxAttempts}",
```
а где настраиваешь эти параметры?

damir2407 commented on scrapper/src/main/java/edu/java/clients/GitHubClient.java:
```diff
@@ -21,6 +24,8 @@ public GitHubClient(WebClient.Builder webClientBuilder, String baseUrl) {
     }
 
     @GetMapping(produces = MediaType.APPLICATION_JSON_VALUE)
+    @Retryable(value = {HttpStatusCodeException.class}, maxAttemptsExpression = "${retry.maxAttempts}",
```
а где настраиваешь эти параметры?

damir2407 commented on scrapper/src/main/java/edu/java/clients/GitHubClient.java:
```diff
@@ -21,6 +24,8 @@ public GitHubClient(WebClient.Builder webClientBuilder, String baseUrl) {
 
     @GetMapping(produces = MediaType.APPLICATION_JSON_VALUE)
+    @Retryable(value = {HttpStatusCodeException.class}, maxAttemptsExpression = "${retry.maxAttempts}",
+               backoff = @Backoff(delayExpression = "${retry.delay}"))
```
также должен быть список кодов статусов, на которые хотим ретраить. Не нашел у тебя

Ltaudato1 committed:
hw10

Ltaudato1 committed:
updated workflow

Ltaudato1 committed:
added jars

Ltaudato1 committed:
ACTUALLY added jars

Ltaudato1 committed:
updated docker-files

Ltaudato1 committed:
temporary fix

Ltaudato1 committed:
temporary fix v2

почему-то именно на гите докер билд вообще не проходит, хотя локально всё работает вообще отлично

Ltaudato1 committed:
refactor hw7

Ltaudato1 committed:
Merge pull request #11 from Ltaudato1/hw10

hw10

Ltaudato1 committed:
Merge pull request #10 from Ltaudato1/hw9

hw9

Ltaudato1 committed:
Merge pull request #9 from Ltaudato1/hw8

hw8 is done
